src/selectedtests/task_mappings/create_task_mappings.py

```diff
@@ -187,23 +186,24 @@ def transform(self) -> List[Dict]:
         task_mappings = []
         for mapping in self.mappings:
             cur_mappings = self.mappings.get(mapping)
-            new_mapping = {
-                "source_file": mapping,
-                "project": self.evergreen_project,
-                "repo": self.repo_name,
-                "branch": self.branch,
-                "source_file_seen_count": cur_mappings.get(SEEN_COUNT_KEY),
-            }
-            new_tasks = []
             builds = cur_mappings.get(TASK_BUILDS_KEY)
-            for build in builds:
-                tasks = builds.get(build)
-                for task in tasks:
-                    new_tasks.append(
-                        {"name": task, "variant": build, "flip_count": tasks.get(task)}
-                    )
-            new_mapping["tasks"] = new_tasks
-            task_mappings.append(new_mapping)
+            if builds:
+                new_mapping = {
+                    "source_file": mapping,
+                    "project": self.evergreen_project,
+                    "repo": self.repo_name,
+                    "branch": self.branch,
+                    "source_file_seen_count": cur_mappings.get(SEEN_COUNT_KEY),
+                }
+                new_tasks = []
```

Comment:
[nit] I think this could be simplified with list comprehensions. Something like this:
```
new_tasks = [
    {"name": task_name, "variant": build_name, "flip_count": flip_count}
    for task_name, flip_count in build.items()
    for build_name, build in builds
]
```

Reply:
I tried this but I think it's harder to read :(
```
new_tasks = [
    {"name": task_name, "variant": build_name, "flip_count": flip_count}
    for build_name, build in builds.items()
    for task_name, flip_count in build.items()
]
```